### PR TITLE
dependency install script and testing release workflow

### PR DESCRIPTION
**Pull request by conorbros**
- Add a script to install dependencies needed for testing on ubuntu
- run the release workflow in tests
- rename some workflows to clearer names

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -50,8 +50,7 @@ jobs:
         key: ${{ matrix.name }}
     - name: Install ubuntu packages
       run: |
-        sudo apt-get update
-        sudo apt-get install -y libpcap-dev
+        shotover-proxy/build/install_ubuntu_packages.sh
```
Comment (rukai) [inline .github/workflows/test.yml:53]: should this go on the same line as the run now?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ on:
 
 jobs:
   publish-image:
+    name: "Publish Docker Image to Docker Hub"
     runs-on: ubuntu-latest
     steps:
       -
```
Comment (rukai) [inline .github/workflows/release.yml:14]: While we're touching this file, should the `name` go on the same line as `-`, thats what the other steps are doing right?

**Review by rukai — APPROVED** [review]
LGTM

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -8,26 +8,24 @@ on:
 
 jobs:
   publish-image:
+    name: "Publish Docker Image to Docker Hub"
     runs-on: ubuntu-latest
     steps:
-      -
-        name: Login to Docker Hub
+      - name: Login to Docker Hub
```
Comment (rukai) [inline .github/workflows/release.yml:14]: .

**Review by XA21X — APPROVED** [review]

**Pull request merged**